client/state/media/actions.js

```diff
@@ -6,21 +6,78 @@ import { castArray } from 'lodash';
 /**
  * Internal dependencies
  */
-import { MEDIA_DELETE, MEDIA_RECEIVE } from 'state/action-types';
+import {
+	MEDIA_DELETE,
+	MEDIA_RECEIVE,
+	MEDIA_REQUEST,
+	MEDIA_REQUEST_FAILURE,
+	MEDIA_REQUESTING } from 'state/action-types';
 
 /**
  * Returns an action object used in signalling that media item(s) for the site
  * have been received.
  *
  * @param  {Number}         siteId Site ID
  * @param  {(Array|Object)} media  Media item(s) received
+ * @param  {Number}         found  Number of found media
+ * @param  {Object}         query  Query Object
  * @return {Object}                Action object
  */
-export function receiveMedia( siteId, media ) {
+export function receiveMedia( siteId, media, found, query ) {
 	return {
 		type: MEDIA_RECEIVE,
+		siteId,
 		media: castArray( media ),
-		siteId
+		found,
+		query
+	};
+}
+
+/**
+ * Returns an action object used in signalling that media item(s) for the site
+ * have been requested.
+ *
+ * @param  {Number} siteId Site ID
+ * @param  {Object} query  Query object
+ * @return {Object}        Action object
+ */
+export function requestMedia( siteId, query ) {
+	return {
+		type: MEDIA_REQUEST,
+		siteId,
+		query
+	};
+}
+
+/**
+ * Returns an action object used in signalling that media item(s) for the site
+ * are being requested.
+ *
+ * @param  {Number} siteId Site ID
+ * @param  {Object} query  Query object
+ * @return {Object}        Action object
+ */
+export function requestingMedia( siteId, query ) {
```

Comment:
Actually there is a difference: `requestMedia` is responsible of triggering the middleware handler. While `requestingMedia` tells that we are effectively starting the Network Request. Because the middleware could decide to avoid triggering a request if we're already requesting the same request.

I guess we'll see more of those while moving to the data middleware